eng/common/scripts/Package-Properties.ps1

```diff
@@ -129,19 +129,30 @@ function Get-PrPkgProperties([string]$InputDiffJson) {
             if ($shouldInclude) {
                 $packagesWithChanges += $pkg
 
-                if ($pkg.DependentPackages) {
-                    $dependentPackagesForInclusion += $pkg.DependentPackages
+                if ($pkg.CanaryPackages) {
+                    $dependentPackagesForInclusion += $pkg.CanaryPackages
                 }
             }
         }
     }
 
-    foreach ($addition in $dependentPackagesForInclusion) {
-        if ($lookup[$addition]) {
-            $packagesWithChanges += $lookup[$addition]
+    # there is a custom function to get dependent packages from a changed package set and diff
+    if ($DependentPackagesFromPackageSetFn -and (Test-Path "Function:$DependentPackagesFromPackageSetFn"))
+    {
+        $packagesWithChanges += &$DependentPackagesFromPackageSetFn $packagesWithChanges $diff
+    }
+    else {
+        foreach ($addition in $dependentPackagesForInclusion) {
```

Comment:
You could remove the variable `$dependentPackagesForInclusion` and pull directly from packages with changes here:

```powershell
foreach ($addition in $packagesWithChanges.CanaryPackages) {
```

Reply:
We'd still need two.

```
foreach ($packageWithChanges in $pkgProps)
  foreach($addition in $packageWithChanges.CanaryPackages)
```

🤷‍♂️